test/middleware.js

```diff
@@ -0,0 +1,212 @@
+var util = require('util');
+var async = require('async');
+var app = require('../app');
+var loopback = require('loopback');
+var ConfigFile = app.models.ConfigFile;
+var Middleware = app.models.Middleware;
+var Facet = app.models.Facet;
+var TestDataBuilder = require('loopback-testing').TestDataBuilder;
+
+describe('Middleware', function() {
+
+  describe('Middleware.create(def, cb)', function() {
+    beforeEach(givenEmptyWorkspace);
+    beforeEach(function(done) {
+      var serverFacet = this.serverFacet;
+      this.configFile = new ConfigFile({
+        path: serverFacet + '/middleware.json'
+      });
+      async.series([function(done) {
+        Middleware.addMiddleware({
+          facetName: serverFacet,
+          name: 'foo',
+          paths: ['/foo'],
+          params: {
+            fooParam: 'foo'
+          },
+          phase: 'routes'
+        }, done);
+      }, function(done) {
+        Middleware.addMiddleware({
+          facetName: serverFacet,
+          name: 'foo-before',
+          paths: ['/foo-before'],
+          params: {
+            barParam: 'foo-before'
+          },
+          phase: 'routes',
+          subPhase: 'before'
+        }, done);
+      }, function(done) {
+        Middleware.addMiddleware({
+          facetName: serverFacet,
+          name: 'bar',
+          paths: ['/bar'],
+          params: {
+            barParam: 'bar'
+          },
+          phase: 'routes'
+        }, done);
+      }, function(done) {
+        Middleware.addMiddleware({
+          facetName: serverFacet,
+          isPhasePlaceHolder: true,
+          phase: 'myPhase'
+        }, done);
+      }, function(done) {
+        Middleware.addMiddleware({
+          facetName: serverFacet,
+          name: 'baz',
+          paths: ['/baz'],
+          params: {
+            barParam: 'baz'
+          },
+          phase: 'auth',
+          nextPhase: 'routes'
+        }, done);
+      }], done);
+    });
+
+    beforeEach(function(done) {
+      this.configFile.load(done);
+    });
+
+    it('should be able to create multiple', function(done) {
```

Comment:
"it should be able to create multiple" - multiple what? phases? middleware entries? something else?